src/main/java/by/exadel/internship/service/impl/UserTimeSlotServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+package by.exadel.internship.service.impl;
+
+
+import by.exadel.internship.dto.timeForCall.UserTimeSlotWithUserDTO;
+import by.exadel.internship.dto.timeForCall.UserTimeSlotWithUserIdDTO;
+import by.exadel.internship.dto.timeForCall.UserTimeSlotDTO;
+import by.exadel.internship.dto.UserDTO;
+import by.exadel.internship.dto.enums.InterviewTime;
+import by.exadel.internship.exception_handing.NotFoundException;
+import by.exadel.internship.mapper.UserTimeSlotMapper;
+import by.exadel.internship.repository.UserTimeSlotRepository;
+import by.exadel.internship.service.UserTimeSlotService;
+import by.exadel.internship.service.UserService;
+import by.exadel.internship.util.MDCLog;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+;
+
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class UserTimeSlotServiceImpl implements UserTimeSlotService {
+
+    private final UserTimeSlotRepository userTimeSlotRepository;
+    private final UserService userService;
+    private final UserTimeSlotMapper mapper;
+
+    private static final int DEFAULT_START_MINUTES_IF_BETWEEN_ZERO_THIRTY = 30;
+    private static final int DEFAULT_START_MINUTES_IF_BETWEEN_THIRTY_ZERO = 0;
+    private static int INTERVIEW_TIME = 30;
+    private static final String SIMPLE_CLASS_NAME = UserTimeSlotService.class.getSimpleName();
+
+
+    @Override
+    public void saveUserTime(UserDTO userDTO) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Try to save free time list for user with uuid = {} in DB", userDTO.getId());
+        List<UserTimeSlotWithUserDTO> resultUSerTimeList = new ArrayList<>();
+        userDTO.getUserTimeSlots().forEach(timeForCallUser -> {
+            checkTime(timeForCallUser);
+            separateTime(timeForCallUser, userDTO.getInterviewTime(), resultUSerTimeList);
+        });
+        resultUSerTimeList.forEach(time -> {
+            time.setUser(userDTO);
```

Comment:
Q: 1. why should entity `UserTimeSlot` hold reference to User, not to userId?
2.  Does `UserTimeSlot` updating affect `User` entity? I mean, if there's any modification with user are they saved on cascade?

Reply:
1 I did reference to User then to do @ManyToOne mapping.
2 I tested, i can only read User

Reply:
>  I did reference to User then to do @manytoone mapping.
I saw, my question was where do we need to send user time slots with user information?

Reply:
I did this because Zhanna asked to make it possible for a person to choose a user and we decided to send information so that a person can choose not only in time but also by user